client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/allocateQuantities.ts

```diff
@@ -0,0 +1,200 @@
+import { DraftStockOutLineFragment } from '../../../api/operations.generated';
+import { canAutoAllocate } from './utils';
+
+/**
+ * Attempts to allocate the requested quantity to the available stock lines.
+ *
+ * @param draftLines - The stock lines to allocate from - should be sorted according to allocation strategy (e.g. FEFO)
+ * @param requestedUnits - The quantity to allocate
+ *
+ * Only available stock lines will be allocated to (i.e. not on hold, not expired, etc).
+ *
+ * How is stock allocated? We try to allocate as close to the requested quantity as possible,
+ * with a bias towards the first stock line(s) in the list (e.g. first expiring).
+ *
+ * This can take up to 3 steps, let's look at an example with 2 stock lines:
+ *
+ * 200 x pack size 1    (exp 01/26)
+ * 3 x   pack size 100  (exp 03/26)
+ *
+ * And we are requesting 350 units.
+ *
+ * Step 1: from top of list allocate to the nearest (rounded down) pack size
```

Comment:
We actually need to confirm this allocation, I think OG uses FEFO full stop, with fractional number of packs if below pack size.Which makes sense, imagine you have a pallet of good expiring this week, and a carton expiring this year, system should really suggest that you split the pallet when issuing a carton